### PR TITLE
[Workers Tutorial]How to use Cloudflare Workers to send form submissions from an Astro+Resend app

### PR DESCRIPTION
**Pull request by CodyDubu**
Contribution to Cloudflare’s Developer Spotlight program.

### CODE REVIEW

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+## 1. Create a new Astro project and install Cloudflare Adapter: 
+
+Open your terminal and run the below command:
+
+```bash
+---
+header: Create Astro project
+---
+npm create cloudflare@latest my-astro-app -- --framework=astro/
```
Comment (megaconfidence) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]: It seems the trailing slash in `--framework=astro/` breaks the CLI

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+Follow the prompts to configure your project, selecting your preferred options for TypeScript usage, TypeScript strictness, version control, and deployment.
+	
+After the initial installation, run the following to add the Cloudflare adapter:
+	
+```bash
+---
+header: Install Cloudflare Adapter
+---
+npm run astro add cloudflare
```
Comment (megaconfidence) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md:40]: I guess readers should run this in the project directory yes?

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+updated: 2024-05-07
+difficulty: Beginner
+content_type: 📝 Tutorial
+pcx_content_type: tutorial
+title: How to use Cloudflare Workers to send form submissions from an Astro+Resend app
+---
+
+# How to use Cloudflare Workers to send form submissions from an Astro+Resend app
+
+This article will instruct5 you on how to send emails from Astro and Cloudflare Workers (via Cloudflare SSR Adapter) using Resend.
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]:
```suggestion
This tutorial will instruct you on how to send emails from [Astro](https://astro.build/) and Cloudflare Workers using [Resend](https://resend.com/).
```

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+After the initial installation, run the following to add the Cloudflare adapter:
+	
+```bash
+---
+header: Install Cloudflare Adapter
+---
+npm run astro add cloudflare
+```
+
+This adapter allows Astro's Server-Side Rendered (SSR) sites and components to work on Cloudflare Pages and converts Astro's endpoints into Cloudflare Workers.
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]:
```suggestion
The [`@astrojs/cloudflare` adapter](https://github.com/withastro/adapters/tree/main/packages/cloudflare#readme) allows Astro's Server-Side Rendered (SSR) sites and components to work on Cloudflare Pages and converts Astro's endpoints into Cloudflare Workers endpoints.
```

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    - Copy and paste the DNS records from Resend to Cloudflare.
+3. **Verify Your Domain:**
+    - Return to Resend and click on the "Verify DNS Records" button.
+    - If everything is set up correctly, your domain status will change to "Verified."
+4. **Create an API Key:**
+    - In Resend, find the "API Keys" option in the side menu and click it.
+    - Create a new API key with a descriptive name and give Full Access permission.
+5. **Save the API key for Local Development and Deployed Worker**
+	- Local Development: create an .env in the root folder of your Astro project and save the API key as RESEND_API_KEY='Api key here' (no quotes).
+	-  Deployed Worker: run the following in your cli and follow the instructions.
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]:
```suggestion
	- For local development, create an .env in the root folder of your Astro project and save the API key as RESEND_API_KEY='Api key here' (no quotes).
	-  For a deployed Worker, run the following in your CLI and follow the instructions.
```

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    <input type="email" id="email" name="email" required />
+    </label>
+    <label>
+    Message
+    <textarea id="message" name="message" required />
+    </label>
+    <button>Send</button>
+</form>
+
+```
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]: I'd recommend running a script formatter on the TypeScript code blocks to make sure the spacing is correct.

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+      status: 404,
+      statusText: "Did not provide the right data",
+    });
+
+  }
+}
+
+```
+
+Next you will need to install the resend SDK.
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]:
```suggestion
Next you will need to install the Resend SDK.
```

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    - In Resend, find the "API Keys" option in the side menu and click it.
+    - Create a new API key with a descriptive name and give Full Access permission.
+5. **Save the API key for Local Development and Deployed Worker**
+	- Local Development: create an .env in the root folder of your Astro project and save the API key as RESEND_API_KEY='Api key here' (no quotes).
+	-  Deployed Worker: run the following in your cli and follow the instructions.
+```bash
+$ npx wrangler secret put RESEND_API_KEY
+```
+
+## 3. Astro Endpoint/Worker
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]: I would change this title to have a call to action, such as 'Create an Astro endpoint'.

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+
+{{<Aside type="warning" header="File Extension">}}
+
+Astro requires an absolute URL, which is why you should use `Astro.url + "/api/sendEmail.json`. If you use a relative path the post request will fail.
+
+{{</Aside>}}
+
+Additionally, adding the `export const prerender = false;` will enable SSR; otherwise, the component will be static and unable to send a post request. If you don't enable it inside the component then you will need to enable SSR via the [template directive](https://docs.astro.build/en/reference/directives-reference/).
+
+After creating the `AstroForm` component, add it to your Astro app, just like any other component [component](https://docs.astro.build/en/basics/astro-components/).
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]:
```suggestion
After creating the `AstroForm` component, add it to your Astro app, just like any other [component](https://docs.astro.build/en/basics/astro-components/).
```

Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md]: I would also recommend showing the reader how to add/import the component to the app they have built. It's a simple step, but will save readers who are less familiar with software development having to click into more documentation to finish this tutorial.

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+    return new Response(null, {
+      status: 404,
+      statusText: "Did not provide the right data",
+    });
+
+  }
+}
+
+```
+
```
Comment (db-cloudflare) [inline content/workers/tutorials/handle-form-submission-with-astro-resend/index.md:108]: The Resend setup could probably be a numbered step in itself to add clarity between creating the Astro endpoint and setting up the Resend functionality. Something such as 'Send emails using Resend' could work.

**Review by db-cloudflare — CHANGES_REQUESTED** [review]
Looks great! I left a few suggestions for your review. Thank you so much for contributing to our docs.

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md (hunk 2 above)
Comment (CodyDubu) [reply content/workers/tutorials/handle-form-submission-with-astro-resend/index.md:40]: Yup, this has been fixed in the recent commit

File: content/workers/tutorials/handle-form-submission-with-astro-resend/index.md (hunk 10 above)
Comment (CodyDubu) [reply content/workers/tutorials/handle-form-submission-with-astro-resend/index.md:108]: Thanks for all the suggestions! I have made the edits in the most recent commit.

**Comment by db-cloudflare** [comment]
LGTM! Great work!

**Review by db-cloudflare — APPROVED** [review]

**Review by db-cloudflare — APPROVED** [review]

**Pull request merged**